packages/venia-ui/lib/components/CheckoutPage/checkoutPage.gql.js

```diff
@@ -64,7 +50,17 @@ export default {
     },
     queries: {
         getCheckoutDetailsQuery: GET_CHECKOUT_DETAILS,
-        getCheckoutStepQuery: GET_CHECKOUT_STEP,
         getOrderDetailsQuery: GET_ORDER_DETAILS
     }
 };
+
+export const checkoutPageResolvers = {
+    Cart: {
+        checkoutStep: cart => {
+            // TODO: Replace with heuristic check against cart data. Requires
+            // fetching more than just total quantity for checkout details query
+            // "cart" arg will have server result.
+            return cart.checkoutStep || 1;
```

Comment:
This seems fine to me right now but if we wanted to get rid of the `onSave` callbacks that we require each component make we could have some heuristic that checks all cart data and sets the current step according to whatever data is missing. Good idea from @tjwiebell :)